fhir-persistence-schema/docs/SchemaDesign.md

```diff
@@ -0,0 +1,650 @@
+
+The `VERSION_HISTORY` table enables tenant version isolation, so upgrades and patches are separately applied to the schema to which the tenant belongs.
+
+The `CreateVersionHistory` class controls the creation the `VERSION_HISTORY` table. Importantly, the VERISON_HISTORY table does not support migrations and changes. The Primary Key has a corresponding index `PK_VERSION_HISTORY`, and the code does not support updating the index. 
+
+This table is an administrative table and should not require updating and migration. 
+
+## Differences Among Database Implementations
+
+The following table highlights the main differences among the database implementations currently supported:
+
+| Database   | Variation |
+| --------   | ----------|
+| DB2        | Multi-tenant. Supports multiple but isolated tenants within one database/schema |
+| DB2        | An SPL stored procedure is used to implement the resource storage logic, reducing the number of application server to database round-trips, improving performance |
+| DB2        | Uses `FHIR_TS` tablespace for admin tables, and a tablespace per tenant |
+| DB2        | `FHIR_TS` is created using a small extent size for efficiency |
+| PostgreSQL | Uses a function for the resource persistence logic |
+| PostgreSQL | TEXT type used instead of CLOB for large data values |
+| Derby      | Resource persistence is implemented at the DAO layer as a sequence of individual statements instead of one procedure call. At a functional level, the process is identical. Simplifies debugging and supports easier unit-test construction. |
+
+----------------------------------------------------------------
+# Schema Design - Physical Data Model
+
+The IBM FHIR Server persists resources using tables defined in a data schema (usually FHIRDATA, but this can be anything). Multiple data schemas can exist in the same database, all managed from a single FHIR_ADMIN schema.
+
+The resource-specific tables are generated by the `FhirResourceTableGroup` class which is called for each value from `FHIRResourceType.ValueSet`.
+
+The schema only stores search values relevant to the most recent resource version.
+
+In addition to tables and indexes, each data schema contains the following sequences:
+
+| Sequence Name | Description |
+|----------|---------------------------------------------|
+| `fhir_sequence` | Provides sequence numbers for the normalized resource ids |
+| `fhir_ref_sequence` | Provides sequence numbers for reference tables such as PARAMETER_NAMES, CODE_SYSTEMS and RESOURCE_TYPES |
+
+These sequences should not require updating and should never be altered as this would risk the integrity of existing data.
+
+By convention, tables are named using the plural form of the data they represent. The following diagram shows the main relationships among tables in the schema. Note that only the search parameter tables for the Patient resource are shown. Each resource type gets its own set of search parameter tables.
```

Comment:
I expected the diagram here but its actually a couple sections below

Reply:
I moved it up